src/Mapping.php

```diff
@@ -573,11 +573,11 @@ private function getBaseData(array $data)
      */
     private function buildColumns()
     {
-        $columns = $this->definition->getPrimaryKey();
+        $columns = $this->prefixTableNameTo($this->definition->getPrimaryKey());
         $required = array_merge($this->definition->getColumns(), $this->columns);
 
         foreach ($this->definition->getProperties() as $item) {
-            $required[] = $item->getLocalColumn();
+            $required[] = in_array($item->getLocalColumn(), $this->definition->getPrimaryKey()) ? $this->prefixTableNameTo($item->getLocalColumn()) : $item->getLocalColumn();
         }
 
         foreach (array_unique($required) as $column) {
```

Comment:
This is already preventing duplicates, however it's possible that the primary key columns which are already included in the `$columns` array are repeated in `$required`. Do we need to do further deduplication?

It's obviously working as it is, but there would be duplicate columns specified in the generated SQL.